### PR TITLE
Refactor configuration and DB tools

### PR DESCRIPTION
**Pull request by jiangxin369**
## What is the purpose of the change

Refactor configuration tools

## Brief change log

*(for example:)*
  - Remove deprecated files
  - Add configuration tools
  - Add database tools

## Verifying this change

*(Please pick either of the following options)*

This change added tests.

### CODE REVIEW

File: ai_flow/common/configuration/config_constants.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+STATE_BACKEND_URI = SERVER_CONF.get_str('state_backend_uri')
+
+SQLALCHEMY_DB_URI = SERVER_CONF.get_str('sql_alchemy_db_uri')
+
+TASK_EXECUTOR = SERVER_CONF.get_str('task_executor', fallback='local')
+
+SQLALCHEMY_POOL_ENABLED = SERVER_CONF.get_bool('sql_alchemy_pool_enabled', fallback=True)
+
+SQLALCHEMY_POOL_SIZE = SERVER_CONF.get_int('sql_alchemy_po ol_size', fallback=5)
```
Comment (wuchaochen) [inline ai_flow/common/configuration/config_constants.py:46]: Rename sql_alchemy_po ol_size to sql_alchemy_pool_size

File: ai_flow/common/configuration/config_constants.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+METADATA_BACKEND_URI = SERVER_CONF.get_str('metadata_backend_uri')
+
+HISTORY_RETENTION = SERVER_CONF.get_str('history_retention')
+
+NOTIFICATION_SERVER_URI = SERVER_CONF.get_str('notification_server_uri')
+
+STATE_BACKEND_URI = SERVER_CONF.get_str('state_backend_uri')
+
+SQLALCHEMY_DB_URI = SERVER_CONF.get_str('sql_alchemy_db_uri')
```
Comment (wuchaochen) [inline ai_flow/common/configuration/config_constants.py:40]: sql_alchemy_db_uri -> sqlalchemy_db_uri

File: ai_flow/common/configuration/config_constants.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+TASK_EXECUTOR = SERVER_CONF.get_str('task_executor', fallback='local')
+
+SQLALCHEMY_POOL_ENABLED = SERVER_CONF.get_bool('sql_alchemy_pool_enabled', fallback=True)
```
Comment (wuchaochen) [inline ai_flow/common/configuration/config_constants.py:44]: sql_alchemy_pool_enabled -> sqlalchemy_pool_enabled

File: ai_flow/common/configuration/configuration.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        return self.as_dict() == Configuration(other).as_dict()
+
+    def __str__(self) -> str:
+        return str({k: v for k, v in sorted(self.as_dict().items())})
+
+
+def get_client_configuration():
+    client_config_file_name = 'aiflow_client.yaml'
+    config_path = os.path.join(get_aiflow_home(), client_config_file_name)
+    if not os.path.isfile(config_path):
```
Comment (wuchaochen) [inline ai_flow/common/configuration/configuration.py:117]: Implement the generation configuration logic and the configuration acquisition separately

File: ai_flow/common/configuration/configuration.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        write_default_config(client_config_file_name)
+    config = Configuration(config_path)
+    return config
+
+
+def get_server_configuration():
+    server_config_file_name = 'aiflow_server.yaml'
+    config_path = os.path.join(get_aiflow_home(), server_config_file_name)
+    if not os.path.isfile(config_path):
```
Comment (wuchaochen) [inline ai_flow/common/configuration/configuration.py:126]: Implement the generation configuration logic and the configuration acquisition separately

File: ai_flow/common/result.py [hunk 6]
```diff
@@ -14,9 +14,13 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+from enum import Enum
 from typing import Text
 
-from ai_flow.common.status import Status
+
+class Status(Enum):
```
Comment (wuchaochen) [inline ai_flow/common/result.py:21]: Don't use the name Status

File: ai_flow/common/status.py [hunk 7]
```diff
@@ -15,10 +14,47 @@
...
 # specific language governing permissions and limitations
 # under the License.
-#
-from enum import Enum
 
 
-class Status(Enum):
-    OK = 0,
-    ERROR = 1,
+class Status:
```
Comment (wuchaochen) [inline ai_flow/common/status.py:19]: Status move to model package.

File: ai_flow/common/util/db_util/session.py [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    pool_max_overflow = config_constants.SQLALCHEMY_MAX_OVERFLOW
+    pool_kwargs = {}
+    if enable_pool:
+        pool_kwargs['pool_size'] = int(pool_size)
+        pool_kwargs['max_overflow'] = int(pool_max_overflow)
+        logger.info("Create SQLAlchemy engine with pool options %s", pool_kwargs)
+    return sqlalchemy.create_engine(db_uri, pool_pre_ping=True, **pool_kwargs)
+
+
+def create_session():
```
Comment (wuchaochen) [inline ai_flow/common/util/db_util/session.py:72]: It need to have db_uri parameter.

File: ai_flow/common/util/db_util/session.py [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+            raise
+        except Exception as e:
+            session.rollback()
+            raise AIFlowDBException(error_msg=e)
+        finally:
+            session.close()
+
+    return make_managed_session
+
+
```
Comment (wuchaochen) [inline ai_flow/common/util/db_util/session.py:56]: Add tests.

File: ai_flow/common/configuration/configuration.py (hunk 4 above)
Comment (jiangxin369) [reply ai_flow/common/configuration/configuration.py:117]: Modified

File: ai_flow/common/result.py (hunk 6 above)
Comment (jiangxin369) [reply ai_flow/common/result.py:21]: Renamed to `RetrunCode`

File: ai_flow/common/util/db_util/session.py (hunk 8 above)
Comment (jiangxin369) [reply ai_flow/common/util/db_util/session.py:72]: Added

**Pull request merged**